libvmi/json_profiles/volatility_ist.c

```diff
@@ -213,6 +213,41 @@ volatility_ist_symbol_to_rva(
     return ret;
 }
 
+status_t
+volatility_build_reverse_symbol_table(json_object *json, GHashTable** table)
+{
+    status_t ret = VMI_FAILURE;
+    json_object *symbols = NULL;
+    if (!json_object_object_get_ex(json, "symbols", &symbols)) {
+        dbprint(VMI_DEBUG_MISC, "Volatility IST profile: no symbols section found\n");
+        goto exit;
+    }
+
+    *table = g_hash_table_new(g_direct_hash, g_direct_equal);
+    struct json_object_iterator it = json_object_iter_begin(symbols);
+    struct json_object_iterator it_end = json_object_iter_end(symbols);
+    for (; !json_object_iter_equal(&it, &it_end); json_object_iter_next(&it)) {
+        json_object  *jsymbol = NULL, *address = NULL;
+        jsymbol = json_object_iter_peek_value(&it);
+        const char* symbol_name = json_object_iter_peek_name(&it);
+        if (!json_object_object_get_ex(jsymbol, "address", &address)) {
+            dbprint(VMI_DEBUG_MISC, "Volatility IST: no address found for %s\n", symbol_name);
+            continue;
+        }
+        addr_t rva;
+#ifdef JSONC_UINT64_SUPPORT
+        rva = json_object_get_uint64(address);
+#else
+        rva = json_object_get_int64(address);
+#endif
+        g_hash_table_insert(*table, GINT_TO_POINTER(rva), (gpointer)symbol_name);
```

Comment:
Please use `GSIZE_TO_POINTER` for values that come from `addr_t`. In this case we are talking about RVAs so it's unlikely INT wrapping would be an issue but still.

Reply:
Changed to `GSIZE_TO_POINTER `